pkg/network/encoding/encoding.go

```diff
@@ -80,27 +80,46 @@ func modelConnections(conns *network.Connections) *model.Connections {
 	dnsFormatter := newDNSFormatter(conns, ipc)
 	tagsSet := network.NewTagsSet()
 
-	for i, conn := range conns.Conns {
-		agentConns[i] = FormatConnection(conn, routeIndex, httpEncoder, http2Encoder, kafkaEncoder, dnsFormatter, ipc, tagsSet)
+	for _, conn := range conns.Conns {
+		builder.AddConns(func(builder *model.ConnectionBuilder) {
+			FormatConnection(builder, conn, routeIndex, httpEncoder, http2Encoder, kafkaEncoder, dnsFormatter, ipc, tagsSet)
+		})
 	}
 
 	routes := make([]*model.Route, len(routeIndex))
 	for _, v := range routeIndex {
 		routes[v.Idx] = &v.Route
 	}
 
-	payload := new(model.Connections)
-	payload.AgentConfiguration = agentCfg
-	payload.Conns = agentConns
-	payload.Domains = dnsFormatter.Domains()
-	payload.Dns = dnsFormatter.DNS()
-	payload.ConnTelemetryMap = FormatConnectionTelemetry(conns.ConnTelemetry)
-	payload.CompilationTelemetryByAsset = FormatCompilationTelemetry(conns.CompilationTelemetryByAsset)
-	payload.KernelHeaderFetchResult = model.KernelHeaderFetchResult(conns.KernelHeaderFetchResult)
-	payload.CORETelemetryByAsset = FormatCORETelemetry(conns.CORETelemetryByAsset)
-	payload.PrebuiltEBPFAssets = conns.PrebuiltAssets
-	payload.Routes = routes
-	payload.Tags = tagsSet.GetStrings()
-
-	return payload
+	builder.SetAgentConfiguration(func(w *model.AgentConfigurationBuilder) {
+		w.SetDsmEnabled(agentCfg.DsmEnabled)
+		w.SetNpmEnabled(agentCfg.NpmEnabled)
+		w.SetUsmEnabled(agentCfg.UsmEnabled)
+	})
+	for _, d := range dnsFormatter.Domains() {
+		builder.AddDomains(d)
+	}
+
+	for _, route := range routes {
+		builder.AddRoutes(func(w *model.RouteBuilder) {
+			w.SetSubnet(func(w *model.SubnetBuilder) {
+				w.SetAlias(route.Subnet.Alias)
+			})
+		})
+	}
+
+	dnsFormatter.FormatDNS(builder)
+
+	for _, tag := range tagsSet.GetStrings() {
+		builder.AddTags(tag)
+	}
+
+	FormatConnectionTelemetry(builder, conns.ConnTelemetry)
+	FormatCompilationTelemetry(builder, conns.CompilationTelemetryByAsset)
+	FormatCORETelemetry(builder, conns.CORETelemetryByAsset)
```

Comment:
does it make sense to export those functions?

Reply:
These were already public, and I was keeping the diff as small as possible, but if you think it's useful, I think these can be made private

Reply:
The diff showed it as newly exported, but you're right
I'm fine with keeping it as it